.github/workflows/build.yml

```diff
@@ -25,28 +29,32 @@ jobs:
         with:
           path: '**/node_modules'
           key: node-modules-${{ hashFiles('**/package-lock.json') }}
-      - name: Cache Cypress
-        id: cache-cypress
-        uses: actions/cache@v3
-        with:
-          path: ~/.cache/Cypress
-          key: cypress-cache
       - name: Cache eslint
+        if: ${{ matrix.task == 'lint' }}
         id: cache-eslint
         uses: actions/cache@v3
         with:
           path: .eslintcache
           key: eslint-${{github.sha}}
+      - name: Cache Cypress
+        if: ${{ matrix.task == 'test:e2e:ci' || matrix.task == 'test:component' }}
+        id: cache-cypress
+        uses: actions/cache@v3
+        with:
+          path: ~/.cache/Cypress
+          key: cypress-cache
       - name: Install lerna and all packages
```

Comment:
To save the `jest` cache, something like this can be done:

```
      - name: Cache Jest
        if: ${{ matrix.task == 'test:unit' }}
        id: cache-jest
        uses: actions/cache@v3
        with:
          path: ./tmp/jest
          key: jest-${{github.sha}}
          restore-keys: jest-
```

For this to work, we should change the cache directory of `jest`.
To do it, I would create a `base-config` on the root of the monorepo, set there the `cacheDirectory` and extend from that base config in all the `jest.config` files.

Something like:

`jest-base.config.js`
```
module.exports = {
  cacheDirectory: '<rootDir>/../../tmp/jest'
};
```

`package/*/jest-config.js`
```
const base = require('../../jest-base.config');

module.exports = {
  ...base,
  // The package specific jest options ...
};
```

Also add the `tmp` directory to the `.gitignore`.